src/onnx/onnx.cpp

```diff
@@ -2486,6 +2487,28 @@ struct onnx_parser
         return l;
     }
 
+    instruction_ref
```

Comment:
After looking back at parse_equal(), parse_greater(), and parse_less(), they are all basically the same except the operator name. You could probably refactor these functions into using just one parse_compare() function, which takes in an additional parameter of op_name (see parse_reduce_oper() function for an example).

You will also need to change L139, L145, and L150 to pass in the second argument of the op_name (see L164-168 for an example).

Reply:
@turneram, the combined function would be like:

    instruction_ref parse_compare_op(const std::string&,
                                 const std::string& op_name,
                                 const node_info&,
                                 std::vector<instruction_ref> args)
    {
        auto l = add_broadcastable_binary_op(args[0], args[1], op_name);
        if(l->get_shape().type() != shape::bool_type)
        {
            l = prog.add_instruction(make_op("convert", {{"target_type", shape::bool_type}}), l);
        }
        return l;
    }

Then the add_mem_op is called like:

add_mem_op("Equal", "equal", &onnx_parser::parse_compare_op);
add_mem_op("Greater", "greater", &onnx_parser::parse_compare_op);
add_mem_op("Less", "less", &onnx_parser::parse_compare_op);